### PR TITLE
Correctly sync shared datasources

### PR DESCRIPTION
**Pull request by philippjfr**
Adds a context manager which ensures that while updating a composite plot any datasources shared among multiple subplots are appropriately synced, avoiding warnings and bokehjs errors. Fixes https://github.com/ioam/holoviews/issues/1135.

### CODE REVIEW

File: holoviews/plotting/bokeh/plot.py [hunk 1]
```diff
@@ -692,13 +698,14 @@ def initialize_plot(self, plots=None, ranges=None):
...
 
-
+    @update_shared_sources
     def update_frame(self, key, ranges=None):
         """
         Update the internal state of the Plot to represent the given
         key tuple (where integers represent frames). Returns this
         state.
         """
+        source_cols = self.handles.get('source_cols', {})
```
Comment (philippjfr) [inline holoviews/plotting/bokeh/plot.py]: Remove

File: tests/testplotinstantiation.py [hunk 2]
```diff
@@ -1224,6 +1224,45 @@ def test_shared_axes_disable(self):
         self.assertEqual((x_range.start, x_range.end), (-.5, .5))
         self.assertEqual((y_range.start, y_range.end), (-.5, .5))
 
+    def test_layout_shared_source_synced_update(self):
+        hmap = HoloMap({i: Dataset({chr(65+j): np.random.rand(i+2)
```
Comment (philippjfr) [inline tests/testplotinstantiation.py]: Add comments and check source_cols is A, B, C, D

**Comment by jlstevens** [comment]
Happy to merge once you've addressed those two comments.

**Comment by philippjfr** [comment]
Addressed the comments, ready to merge when tests pass.

**Comment by jlstevens** [comment]
Had to restart the tests due to transients but they are all passing now. Merging.

**Pull request merged**